packages/graphback-codegen-client/src/helpers/writeDocuments.ts

```diff
@@ -1,17 +1,31 @@
-import { existsSync, mkdirSync, writeFileSync } from 'fs';
+import { existsSync, mkdirSync, writeFileSync, PathLike, MakeDirectoryOptions } from 'fs';
 import { ClientTemplates } from '../templates/ClientTemplates';
 
-export const writeDocumentsToFilesystem = (clientPath: string, generatedClient: ClientTemplates, extension: 'ts' | 'graphql') => {
-    if (!existsSync(clientPath)) {
-        mkdirSync(clientPath, { recursive: true })
-    }
-    Object.keys(generatedClient).forEach((folder: string) => {
-        const currentFolder = `${clientPath}/${folder}`
-        if (!existsSync(currentFolder)) {
-            mkdirSync(currentFolder)
+interface IWriter {
```

Comment:
Why we need this?

Reply:
This is always going to run on Node.js API so we can hardcode those

Reply:
So I can inject my own functions for testing

Reply:
Not needed. You can dissengage code that creates string from creation of the file. Creation of the file code do not require testing etc. For the rest we use snapshots effectively